### ISSUE TITLE
timeouts on jenkins for GitHub builds

### ISSUE THREAD

**Issue by michalmac** [matsim-org/matsim-libs#1289]
We are often having almost 3-hour builds. This one was about to end in 1-2 minutes:
http://ci.matsim.org:8080/job/MATSim_GitHub/3436/console
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:58 h
[INFO] Finished at: 2020-12-12T00:25:07+01:00
[INFO] ------------------------------------------------------------------------
Waiting for Jenkins to finish collecting data
Build timed out (after 180 minutes). Marking the build as aborted.
Build timed out (after 180 minutes). Marking the build as failed.
Aborted
```
Maybe we should add 30 min more?

**Comment by mrieser** [comment]
I increased the timeout to 3.5 hours. But we should probably also check why it takes so long. For example, it currently runs the normal tests plus the integration tests. Is this really necessary, or can we speed up the builds otherwise?

**Comment by michalmac** [comment]
>For example, it currently runs the normal tests plus the integration tests. Is this really necessary, or can we speed up the builds otherwise?

Re running ITs: Probably depends on what is the purpose of these builds

Not sure if I recall it correctly, but the story behind this build pipeline was: It was introduced during the Dev Meet in Estonia 2018. We had some issues with travis (maybe just too many pushes). So the builds on Jenkins were added, but (at some point?) they only run unit tests. That reduced the build time to less than 30 min and helped us to keep pushing and merging (not so well tested -- some may say) code. At some point in time, we included ITs to the github builds (to improve the test coverage), but then build times increased to 2.5 hours.

A good thing about the github builds is that they run mvn for the whole matsim project (not just matsim (main) + contribs), which we do not do in travis / github actions (but we do in appveyor).

===

Some general ideas (apart from parallelising mvn builds):
- fewer contribs (remove old, non-maintained and unused ones)
- test cleanup/refactoring: many heavy ITs (easier to add) could be changed to compact unit tests (easier to maintain)
